admin/server/github.go

```diff
@@ -1069,20 +1163,37 @@ func (s *Server) pushToGit(ctx context.Context, copyData func(projPath string) e
 		return fmt.Errorf("failed to add files to git: %w", err)
 	}
 
+	var sign *object.Signature
 	client := github.NewTokenClient(ctx, token)
 	user, _, err := client.Users.Get(ctx, "")
 	if err != nil {
-		return fmt.Errorf("failed to get current user: %w", err)
+		// fallback to username and email in Rill
+		claims := auth.GetClaims(ctx)
+		if claims.OwnerType() != auth.OwnerTypeUser {
+			// ideally should never reach here
+			return fmt.Errorf("failed to get current user: %w", err)
+		}
+		u, err := s.admin.DB.FindUser(ctx, claims.OwnerID())
+		if err != nil {
+			return err
+		}
```

Comment:
Consider doing this in the RPC implementation and passing the desired name/email as parameters to this function. That way `pushToGit` is not coupled to server middleware, and can be considered a standalone util function, which may in the future e.g. be tested stand-alone or invoked from a background job instead.